proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,91 @@
+# ***Add `nuget search` Functionality to DotNet***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The specification seeks to add search functionality to dotnet. The functionality will be added to dotnet as a search command: `dotnet package search`.  
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+As outlined in the customer request on this GitHub [issue](https://github.com/NuGet/Home/issues/6060), there is a call for the incorporation of `nuget.exe list` functionality into dotnet. The customers are requesting for this feature, but a `search` command can replicate the same functionality, with the only difference being the quantity of package results it outputs. A search command mitigates unnecessary server use by limiting the number of results through the `--take` option. By integrating this search functionality, we aim to accommodate our customers' needs, especially for scripting purposes.
```

Comment:
Putting each sentence on a separate line makes it easier to comment on a specific line. Also, this is being completely pedantic, I feel like the focus should not be on the `nuget list` command, but brining the dotnet cli closer to feature parity of nuget.exe. For better or worse, NuGet/Home#6060 is being used to track both the list and search commands (see https://github.com/NuGet/Home/issues/4855#issuecomment-1205936210)

```suggestion
The `dotnet` CLI is missing a way for customers to search for packages from the command line, as is possible with `nuget.exe search`.
This brings NuGet experiences in the `dotnet` CLI one step closer towards feature parity.

While [NuGet/Home#6060](https://github.com/NuGet/Home/issues/6060) (the linked issue) also discusses the `nuget.exe list` command, the issue was created before `nuget.exe search` was added.
The [NuGet Server API Search Resource specification requires all packages to be returned when no `q` parameter is provided](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource#request-parameters).
Therefore, this proposal does not add a `list` equivalent to the `dotnet` CLI, as it is effectively redundant.
```